src/main/java/com/hillel/items_exchange/service/UserService.java

```diff
@@ -127,4 +142,27 @@ private boolean checkReadOnlyFields(User toCompare, User original, String fieldN
         declaredField.setAccessible(true);
         return declaredField.get(toCompare).equals(declaredField.get(original));
     }
-}
\ No newline at end of file
+
+    private boolean isNewUser(User user) {
+        return user.getUpdated().equals(user.getCreated());
+    }
+
+    private <T> Collection<T> extractAll(Collection<T> src, Predicate<T> predicate,
+                                         Supplier<Collection<T>> collectionFactory) {
+        Collection<T> extracted = src.stream()
+                .filter(predicate)
+                .collect(Collectors.toCollection(collectionFactory));
+        src.removeAll(extracted);
+        return extracted;
+    }
+
+    private void addNewChildren(User user, Collection<Child> children) {
+        children.forEach(child -> child.setUser(user));
```

Comment:
Because the first query to the db is not UPDATE of a parent entity, so method with annotation `PreUpdate` isn't invoked,  the first query is INSERT new entities that don't contain user to bidirectional relationships, the query has `null` instead of a user id.